vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/api/get_parameters/get_parameters.cc

```diff
@@ -129,6 +130,19 @@ type::Result GetParameters::ParseBody(
     parameters->monthly_tip_choices.push_back(choice.GetDouble());
   }
 
+  const auto* payout_status_dict = value->GetDict().FindDict("payoutStatus");
+  if (!payout_status_dict) {
+    BLOG(0, "Missing payout status");
+    return type::Result::LEDGER_ERROR;
+  }
+
+  for (const auto [key, value] : *payout_status_dict) {
```

Comment:
I see, the dict iterator constructs a pair when dereferenced, whose lifetime gets extended by the destructured binding, and we have a compiler setting on to complain about that.

Does `for (const auto& pair : *payout_status_dict)` work?

Reply:
Also curious if `for (const auto&& [key, value] : *payout_status_dict)` works :)

Reply:
Yes that does work, perfect!

Reply:
Ha, that's super weird. That's because `base::detail::const_dict_iterator` is a proxy iterator, and Clang warns you that you're extending the lifetime of a `std::pair<const std::string&, const base::Value &>` temporary. Not sure why it doesn't complain for ` const auto&&`, as you're still extending the lifetime... let's remove this `const`, though, since it doesn't convey too much semantic meaning here. 🙂
Also, you can use `emplace<>()`:
```
for (auto&& [key, value] : *payout_status_dict) {
  if (value.is_string()) {
    parameters->payout_status.emplace(key, value.GetString());
  }
}
```

Reply:
Thanks, I always forget about `emplace`!

Reply:
Regarding the `&&` - it's a little cryptic, but I suppose it means that we're extending lifetimes explicitly instead of implicitly, which is better?

Reply:
@emerick @zenparsing FYI: https://github.com/brave/brave-core/pull/14176#discussion_r926889811